flaky: coderd/prometheusmetrics TestWorkspaces/Multiple

Spotted in: https://github.com/coder/coder/actions/runs/3630546787/jobs/6124020920

```
=== FAIL: coderd/prometheusmetrics TestWorkspaces/Multiple (5.00s)
    prometheusmetrics_test.go:227: 
        	Error Trace:	/home/runner/work/coder/coder/coderd/prometheusmetrics/prometheusmetrics_test.go:227
        	            				/home/runner/work/coder/coder/coderd/prometheusmetrics/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestWorkspaces/Multiple
        	Messages:   	invalid count for failed
    prometheusmetrics_test.go:212: 
        	Error Trace:	/home/runner/work/coder/coder/coderd/prometheusmetrics/prometheusmetrics_test.go:212
        	Error:      	Condition never satisfied
        	Test:       	TestWorkspaces/Multiple
    --- FAIL: TestWorkspaces/Multiple (5.00s)
```

--- a/coderd/prometheusmetrics/prometheusmetrics_test.go
+++ b/coderd/prometheusmetrics/prometheusmetrics_test.go
@@ -224,7 +224,9 @@ func TestWorkspaces(t *testing.T) {
 					if !ok {
 						t.Fail()
 					}
-					require.Equal(t, count, int(metric.Gauge.GetValue()), "invalid count for %s", metric.Label[0].GetValue())
+					if metric.Gauge.GetValue() != float64(count) {
+						return false
+					}
 					sum += int(metric.Gauge.GetValue())
 				}
 				t.Logf("sum %d == total %d", sum, tc.Total)

